site/en/docs/extensions/reference/scripting/index.md

```diff
@@ -2,9 +2,191 @@
+[webNavigation API](webNavigation).
+
+```js
+const tabId = getTabId();
+const frameIds = [frameId1, frameId2];
+chrome.scripting.executeScript(
+    {
+      target: {tabId: tabId, frameIds: frameIds},
+      files: ['script.js'],
+    },
+    () => { ... });
+```
+
+Note that you cannot specify both the `frameIds` and `allFrames` properties.
+
+### Injected Code
+
+Extensions can specify the code to be injected either via an external file or a
+runtime variable.
+
+#### Files
+
+Files are specified as strings that are paths relative to the extension's root
+directory. The following code will inject the file `script.js` into the main
+frame of the tab.
+
+```js
+const tabId = getTabId();
+chrome.scripting.executeScript(
+    {
+      target: {tabId: tabId},
+      files: ['script.js'],
+    },
+    () => { ... });
+```
+
 !!!.aside.warning
-This documentation is in active development.
-We're planning to expand the introduction section to provide more context and guidance on how to use this API.
-This work is tracked in issue [#194](https://github.com/GoogleChrome/developer.chrome.com/issues/194).
+Note: Currently, a maximum of a single file is supported.
 !!!
 
+#### Runtime Functions
+
+When injecting JavaScript with `scripting.executeScript()`, you can specify a
+function to be executed instead of a file. This function should be a function
+variable available to the current extension context.
+
+```js
+function getTitle() {
+  return document.title;
+}
+const tabId = getTabId();
+chrome.scripting.executeScript(
+    {
+      target: {tabId: tabId},
+      function: getTitle,
+    },
+    () => { ... });
+```
+
+This function will be executed in the context of injection target. However,
+this will not carry over any of the current execution context of the function.
+As such, bound parameters (including the `this` object) and
+externally-referenced variables will result in errors.  For instance, the
+following code will not work, and will throw a ReferenceError because `color`
+is undefined when the function executes:
+
+```js
+const color = getUserColor();
+function changeBackgroundColor() {
+  document.body.style.backgroundColor = color;
+}
+const tabId = getTabId();
+chrome.scripting.executeScript(
+    {
+      target: {tabId: tabId},
+      function: changeBackgroundColor,
+    },
+    () => { ... });
+```
+
+You can work around this by using the [Storage API](storage) or by
+[passing messages](messaging).
+
+The Chromium team is working on providing more capabilities in the future.
+
+#### Runtime Strings
+
+If injecting CSS within a page, you can also specify a string to be used in the
+`css` property. Note that this option is only available for
+`scripting.insertCSS()`; you cannot execute a string via
+`scripting.executeScript()`.
+
+```js
+const css = 'body { background-color = "red"; }';
+const tabId = getTabId();
+chrome.scripting.insertCSS(
+    {
+      target: {tabId: tabId},
+      css: css,
```

Comment:
nit: you can write this as just `css,`

Reply:
yep, same for tabId, too.  I personally find this a bit more clear (though longer).  I'm not sure if that's a common sentiment, or me being old ; )  Simeon or Andy, any preference?

Reply:
Correct - using object declaration shorthand, the property is assigned to that variable's value.  This has been around for awhile, but I don't see it as much, and don't find it quite as clear.  But I'm not sure if that's just me not writing enough JS code ; )

Reply:
I think the way it is at lines 156-157 here is consistent with the way we're doing this elsewhere in the docs.  Fine as is.